svc-events/events/subscriptionCache.go

```diff
@@ -0,0 +1,210 @@
+	emptyOriginResourceToSubscriptionsMap = make(map[string]map[string]bool)
+	subscriptions, err := evmodel.GetAllEvtSubscriptions()
+	if err != nil {
+		l.Log.Error("Error while reading all subscription data ", err)
+		return
+	}
+	for _, subscription := range subscriptions {
+		var sub evmodel.SubscriptionResource
+		err = json.Unmarshal([]byte(subscription), &sub)
+		if err != nil {
+			continue
+		}
+		loadSubscriptionCacheData(sub)
+	}
+
+}
+
+func getAllDeviceSubscriptions() {
+	eventSourceToManagerIDMap = make(map[string]string)
+	t := time.Now()
+	defer l.Log.Debug("Time take to read complete aggregateToHost ", time.Since(t))
+	deviceSubscriptionList, err := evmodel.GetAllDeviceSubscriptions()
+	if err != nil {
+		l.Log.Error("Error while reading all aggregate data ", err)
+		return
+	}
+	for _, device := range deviceSubscriptionList {
+		devSub := strings.Split(device, "||")
+		if strings.Contains(devSub[0], "Collection") {
+			continue
+		} else {
+			updateCatchDeviceSubscriptionData(devSub[0], evmodel.GetSliceFromString(devSub[2]))
+		}
+	}
+}
+func updateCatchDeviceSubscriptionData(key string, originResources []string) {
+	systemId := originResources[0][strings.LastIndexByte(originResources[0], '/')+1:]
+	eventSourceToManagerIDMap[key] = systemId
+}
+
+func loadSubscriptionCacheData(sub evmodel.SubscriptionResource) {
+	if len(sub.EventDestination.OriginResources) == 0 && sub.SubscriptionID != "0" {
+		subCache := evmodel.Subscription{
+			Id:                   sub.SubscriptionID,
+			Destination:          sub.EventDestination.Destination,
+			EventTypes:           sub.EventDestination.EventTypes,
+			MessageIds:           sub.EventDestination.MessageIds,
+			SubordinateResources: sub.EventDestination.SubordinateResources,
+			ResourceTypes:        sub.EventDestination.ResourceTypes,
+			SubscriptionType:     sub.EventDestination.SubscriptionType,
+			OriginResources:      sub.EventDestination.OriginResources,
+			DeliveryRetryPolicy:  sub.EventDestination.DeliveryRetryPolicy,
+		}
+		addEmptyOriginSubscriptionCache(subCache.Id)
+		subscriptionsCache[subCache.Id] = subCache
+	} else {
+		for _, host := range sub.Hosts {
+			subCache := evmodel.Subscription{
+				Id:                   sub.SubscriptionID,
+				Destination:          sub.EventDestination.Destination,
+				EventTypes:           sub.EventDestination.EventTypes,
+				MessageIds:           sub.EventDestination.MessageIds,
+				SubordinateResources: sub.EventDestination.SubordinateResources,
+				ResourceTypes:        sub.EventDestination.ResourceTypes,
+				SubscriptionType:     sub.EventDestination.SubscriptionType,
+				OriginResources:      sub.EventDestination.OriginResources,
+				DeliveryRetryPolicy:  sub.EventDestination.DeliveryRetryPolicy,
+			}
+			addSubscriptionCache(host, subCache.Id)
+			subscriptionsCache[subCache.Id] = subCache
+		}
+	}
+}
+
+//addSubscriptionCache add subscription in corresponding cache based on key type
```

Comment:
Why cant we fill this in bulk and avoid make/extend of the maps??

Reply:
done